File: serene/src/Serene.Web/Modules/ServerTypes/Administration/RoleColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { RoleRow } from "./RoleRow";

export interface RoleColumns {
    RoleId: Column<RoleRow>;
    RoleName: Column<RoleRow>;
}

export class RoleColumns extends ColumnsBase<RoleRow> {
    static readonly columnsKey = 'Administration.Role';
    static readonly Fields = fieldsProxy<RoleColumns>();
}

File: serene/src/Serene.Web/Modules/Administration/Role/RoleColumns.cs
namespace Serene.Administration.Forms;

[ColumnsScript("Administration.Role")]
[BasedOnRow(typeof(RoleRow), CheckNames = true)]
public class RoleColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int RoleId { get; set; }
    [EditLink, Width(300)]
    public string RoleName { get; set; }
}